### PR TITLE
[UI] Filter support for Meshsync and Connections

### PR DESCRIPTION
**Pull request by sudhanshutech**
**Notes for Reviewers**

This PR fixes #

https://github.com/meshery/meshery/assets/90546692/e46b7c2f-a946-4618-b34d-18b23002bead

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

**Comment by sudhanshutech** [comment]
@theBeginner86 
`http://localhost:3000/api/system/meshsync/resources?clusterIds=%5B%2218702e49-8be4-46d9-8f1b-12929a9f5f47%22%2C%22d559042b-f84f-4db3-97d1-4720caa98850%22%5D&page=0&pagesize=0&search=Kubernetes&order=`
if i search anything we get just one error of response of `no such column: kubernetes_resources.name`. I checked doc and handler its saying we can search component name    but same error and there is no filter specifically mentioned

File: ui/utils/custom-filter.js [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+    boxShadow: `0px 4px 8px ${
+      theme.palette.type === 'light'
+        ? theme.palette.secondary.disabledIcon
+        : '0px 4px 5px 0px rgba(0,0,0,0.14)'
+    }`,
+  },
+  icon: {
+    color: theme.palette.secondary.iconMain,
+  },
+});
```
Comment (Yashsharma1911) [inline ui/utils/custom-filter.js:32]: Styled component would be nice

**Review by Yashsharma1911 — CHANGES_REQUESTED** [review]

**Comment by theBeginner86** [comment]
> @theBeginner86 `http://localhost:3000/api/system/meshsync/resources?clusterIds=%5B%2218702e49-8be4-46d9-8f1b-12929a9f5f47%22%2C%22d559042b-f84f-4db3-97d1-4720caa98850%22%5D&page=0&pagesize=0&search=Kubernetes&order=` if i search anything we get just one error of response of `no such column: kubernetes_resources.name`. I checked doc and handler its saying we can search component name but same error and there is no filter specifically mentioned

I see. Looking into these now...

**Comment by theBeginner86** [comment]
> @theBeginner86 `http://localhost:3000/api/system/meshsync/resources?clusterIds=%5B%2218702e49-8be4-46d9-8f1b-12929a9f5f47%22%2C%22d559042b-f84f-4db3-97d1-4720caa98850%22%5D&page=0&pagesize=0&search=Kubernetes&order=` if i search anything we get just one error of response of `no such column: kubernetes_resources.name`. I checked doc and handler its saying we can search component name but same error and there is no filter specifically mentioned

Fixed: https://github.com/meshery/meshery/pull/9737

**Comment by sudhanshutech** [comment]
> > @theBeginner86 `http://localhost:3000/api/system/meshsync/resources?clusterIds=%5B%2218702e49-8be4-46d9-8f1b-12929a9f5f47%22%2C%22d559042b-f84f-4db3-97d1-4720caa98850%22%5D&page=0&pagesize=0&search=Kubernetes&order=` if i search anything we get just one error of response of `no such column: kubernetes_resources.name`. I checked doc and handler its saying we can search component name but same error and there is no filter specifically mentioned
> 
> Fixed: #9737

Thanks, i will update the ui

File: ui/utils/custom-filter.js (hunk 1 above)
Comment (sudhanshutech) [reply ui/utils/custom-filter.js:32]: styled components are messing with the theme currently , so when we work on v5 these all can be converted

**Comment by sudhanshutech** [comment]
Done from my side.

- [x] Fetching kinds from new api in meshsync
- [x] Connections are using param status, kind for filtering

**Comment by sudhanshutech** [comment]
Moving forward here

**Pull request merged**